server/src/test/java/org/elasticsearch/action/ActionListenerTests.java

```diff
@@ -454,6 +457,63 @@ public void onFailure(Exception e) {
         assertThat(exReference.get(), instanceOf(IllegalStateException.class));
     }
 
+    public void testRunBeforeThrowsAssertionErrorIfExecutedMoreThanOnce() {
+        assumeTrue("test only works with assertions enabled", Assertions.ENABLED);
```

Comment:
Huh TIL it's possible to run tests with assertions disabled! I wouldn't be surprised if this didn't work in lots of other ways too tho.

Reply:
Yep...